packages/react-components/review/react-components.api.md

```diff
@@ -165,6 +168,30 @@ export interface GridLayoutProps {
 // @public (undocumented)
 export type GridLayoutType = 'standard';
 
+// @public (undocumented)
+export const IdentifierContext: React_2.Context<Identifiers>;
+
+// @public (undocumented)
+export const IdentifierProvider: (props: IdentifierProviderProps) => JSX.Element;
```

Comment:
I'm fully onboard with:
* Component props for injecting IDs.
* Composite prop to inject IDs for all used components (similar to the `strings` prop).

I'm not yet convinced about the need to add a `Provider` for these though. Unlike internationalization, there is no need for Contoso to add identifiers exhaustively for all components, so let's just stick to the props for now?

For implementation in the Composite, you can define the `IdentifierProvider` inside the composite implementation to avoid having to drill through props. Then, at the point where the composite instnatiates the components, you would have to use the provider and set props for the component.

Reply:
We can always add the `IdentifierProvider` to the API in the future if we see a business need. Revese is not possible.

Reply:
That won't work.
Assigning an ID to `<ParticipantList data-ui-id="someid" />` won't actually assign the element an id.
For that you need to assign it to the `<Stack />` being used inside the component. 
Which means that the only way to inject such ids into components through composites is to expose another component prop that does this for us. That increases the API surface even more...

Reply:
I think Provider should be ok for this case because the context is not expected to change much (or at all) during run time. So rerendering due to IDS context should not be an issue.